crates/rpc/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -103,7 +112,17 @@ impl Transaction {
         let (gas_price, max_fee_per_gas) = match signed_tx.tx_type() {
             TxType::Legacy => (Some(U128::from(signed_tx.max_fee_per_gas())), None),
             TxType::EIP2930 => (None, Some(U128::from(signed_tx.max_fee_per_gas()))),
-            TxType::EIP1559 => (None, Some(U128::from(signed_tx.max_fee_per_gas()))),
+            TxType::EIP1559 => {
+                // the gas price field for EIP1559 is set to `min(tip, gasFeeCap - baseFee) +
+                // baseFee`
+                let gas_price = base_fee
+                    .and_then(|base_fee| {
+                        signed_tx.effective_tip_per_gas(base_fee).map(|tip| tip + base_fee as u128)
+                    })
+                    .unwrap_or_else(|| signed_tx.max_fee_per_gas());
+
+                (Some(U128::from(gas_price)), Some(U128::from(signed_tx.max_fee_per_gas())))
+            }
```

Comment:
Had to write this out to be sure that it computed the `min(max_priority_fee_per_gas + base_fee, max_fee_per_gas)` formula that geth computes

If `base_fee <= max_priority_fee_per_gas`:
```
effective_tip_per_gas = min(max_priority_fee_per_gas, max_fee_per_gas - base_fee)
```
So in the `map` this computes:
```
min(max_priority_fee_per_gas, max_fee_per_gas - base_fee) + base_fee
```

Which is equal to:
```
min(max_priority_fee_per_gas + base_fee, max_fee_per_gas)
```